lib/servers/Pulser2/pulser_ok.py

```diff
@@ -107,6 +109,7 @@ def programSequence(self, c, sequence):
         #print "program sequence"
         sequence = c.get('sequence')
         if not sequence: raise Exception("Please create new sequence first")
+        self.programmed_sequence = sequence
         dds,ttl = sequence.progRepresentation()
```

Comment:
question, self.initialize board gets called by initserver before any context creates a pulse sequence. What does c.get('sequence') return in this case

Reply:
Actually thinking more this can all be bypassed by putting this in 'program sequence' function bypassing this corner case

Reply:
`c.get('sequence')` returns None if program sequence is called before a new sequence is created. If this happens, it will be caught by https://github.com/CampbellGroup/common/pull/215/files/26efa2b36c086891bb4b6a3156852a7b147b7858#diff-a99321155b01b04b828c84762b9742a2R111 as a raised Exception, and self.programmed_sequence remains None.

Reply:
ok, thats fine but i think it would be cleaner if it was in programSequence (line 103)  this would avoid the try and except and is closer to when the pulser actually get programmed

Reply:
`self.programmed_sequence = sequence` (edited line 112) is in programSequence (edited line 105). Do you mean that edited line 112 should be moved to the end of programSequence?